bokehjs/src/coffee/models/tools/draw/poly_draw_tool.ts

```diff
@@ -0,0 +1,105 @@
+import * as p from "core/properties"
+import {SelectTool} from "../gestures/select_tool"
+import {EditToolView} from "./edit_tool"
+import {ColumnDataSource} from "../../sources/column_data_source"
+
+
+export interface BkEv {
+  bokeh: {
+    sx: number
+    sy: number
+  }
+  srcEvent: {
+    shiftKey?: boolean
+  }
+  keyCode: number
+  timeStamp: number
+}
+
+
+export class PolyDrawToolView extends EditToolView {
+  model: PolyDrawTool
+
+  _tap(e: BkEv) {
+    let did_hit;
+    const ds = this.model.source;
+    const indices = ds.selected['1d'].indices.slice(0);
+    const append = e.srcEvent.shiftKey != null ? e.srcEvent.shiftKey : false;
+    did_hit = this._select_event(e, append);
+  }
+
+  _keyup(e: BkEv) {
+    if ((e.keyCode === 8) && this.model.active) {
+      const ds = this.model.source;
+      const { indices } = ds.selected['1d'];
+      indices.sort((a: number, b: number) => a-b);
```

Comment:
This `cmp` function seems unnecessary. `indices.sort()` will achieve the same thing.